### ISSUE TITLE
netcore mono crashes during AppContext.Setup with roslyn 3.3.1-beta3-19423-01 because GenericEqualityComparer1 is no longer GTD

### ISSUE THREAD

**Issue by akoeplinger** [mono/mono#16541]
## Steps to Reproduce

1.  Apply https://github.com/mono/mono/pull/16366
2. Build netcore Mono: `cd netcore && ./build.sh`
3. Run any test, e.g. `cd netcore && make run-tests-corefx-System.Runtime.Tests`

### Current Behavior

It crashes during AppContext.Setup when creating the ` s_dataStore = new Dictionary<string, object?>(count);`.

Digging deeper we found out that it hits an error in `mono_reflection_bind_generic_parameters ` when processing GenericEqualityComparer: https://github.com/mono/mono/blob/2152f0a16d68021fa28a997806992ad59fd0c7e7/mono/metadata/reflection.c#L2546-L2550

A diff of the System.Private.CoreLib.dll between old and new Roslyn looks like this:
https://gist.github.com/akoeplinger/32722e677785730e9745c623d30eedd7

### Expected Behavior

No crash.

## On which platforms did you notice this

[ x ] macOS
[ x ] Linux
[ x ] Windows

**Version Used**: master

**Comment by lambdageek** [comment]
Seems like dotnet/roslyn#38344 is a similar issue outside of System.Private.CoreLib and affecting more than just `typeof (Nullable<>)`

**Comment by lambdageek** [comment]
Confirmed that if I build roslyn from source that includes https://github.com/dotnet/roslyn/pull/38249 (I used dotnet/roslyn@b5260300ca54cfec1419722afed8d05612b909d0) and then use that to overwrite what's in `netcore/roslyn/packages/microsoft.net.compilers.toolset/3.3.1-beta3-19423-01/tasks/netcoreapp2.1` and then build the BCL using that hacked up compiler.

Outcome:

1. `System.Collections.Generic.Comparer<T>::CreateComparer` has the expected ``ldtoken class System.Nullable`1``
2. `cd netcore && make run-sample` produces the epxected output
3. `cd netcore && make run-tests-corefx-System.Runtime.Tests` passes


So I guess we need a build of Roslyn that includes https://github.com/dotnet/roslyn/commit/98b82ec70a2d9e6953a588464bce44fa245182b3
